### PR TITLE
Fix possible division by zero by clamping latitude to 85.05°

### LINKED ISSUE #3530: Div-by-zero in web mercator
With LLVM 3.9.1's ubsan:

```
./unit_tests/util-tests
include/util/web_mercator.hpp:45:69: runtime error: division by zero
```

Needs to be investigated.

### PR DESCRIPTION
**Pull request by oxidase**
# Issue

Resolves #3530 by pre clamping of the input latitude.

## Tasklist
 - [x] review
 - [x] adjust for comments

### CODE REVIEW

File: include/util/web_mercator.hpp [hunk 1]
```diff
@@ -41,10 +41,11 @@ inline FloatLatitude yToLat(const double y)
 inline double latToY(const FloatLatitude latitude)
 {
     // apparently this is the (faster) version of the canonical log(tan()) version
-    const double f = std::sin(detail::DEGREE_TO_RAD * static_cast<double>(latitude));
-    const double y = detail::RAD_TO_DEGREE * 0.5 * std::log((1 + f) / (1 - f));
-    const auto clamped_y = std::max(-180., std::min(180., y));
-    return clamped_y;
+    const double epsg3857_limit = 85.051128779806592378; // 90(4 atan(exp(pi)) / pi - 1)
```
Comment (TheMarex) [inline include/util/web_mercator.hpp]: What about making this constant a `constexpr`? Like `static const constexpr double EPSG3857_MAX_LAT = 85.051128779806592378; // 90(4 atan(exp(pi)) / pi - 1);`

File: unit_tests/util/coordinate_calculation.cpp [hunk 2]
```diff
@@ -118,23 +118,20 @@ BOOST_AUTO_TEST_CASE(compute_angle)
...
     // Invalid values
-    /* TODO: Enable this when I figure out how to use BOOST_CHECK_THROW
-     *       and not have the whole test case fail...
-    first = Coordinate(FloatLongitude{0}, FloatLatitude{0});
-    middle = Coordinate(FloatLongitude{1}, FloatLatitude{0});
-    end = Coordinate(FloatLongitude(std::numeric_limits<double>::max()), FloatLatitude{0});
-    BOOST_CHECK_THROW( coordinate_calculation::computeAngle(first,middle,end),
-                       boost::numeric::positive_overflow);
-                       */
+    BOOST_CHECK_THROW(
```
Comment (TheMarex) [inline unit_tests/util/coordinate_calculation.cpp]: Good catch!

**Comment by oxidase** [comment]
Rebased and changed `MAX_LATITUDE` to `EPSG3857_MAX_LATITUDE`. It influences also `xyzToMercator` where latitude also should be clamped by EPSG:3857 max latitude value and not 85

**Comment by oxidase** [comment]
Update `web_mercator_tests/xyz_to_mercator` test expectations: previous values 19971868.8804085782 and and 19971868.880408578  mean 0 tile y-extent in Mercator projection. Correct maxy value is 6378137*pi or 20037508.342789277 in Mercator projection or 85.0511 in WGS

**Pull request merged**